### PR TITLE
blog: add Farewell to Node.js v5, Preparing for v7 post

### PR DESCRIPTION
**Pull request by rvagg**
@nodejs/lts @nodejs/ctc @nodejs/website as per a commitment I made at an LTS meeting a while back, I was supposed to have this post up just after v5 went EOL but it was never quite completed and posted. Now we're in the in-between limbo-land so I've tried to reframe it as a v5-to-v7 story. It's basically taking the opportunity to do a refresher on the LTS scheduled.

I'd like some feedback from y'all about (a) whether this post makes enough sense to post now (I obviously think it does but the content has floating between my devices and in my head for a couple of months!) and (b) the content itself.

It's got the simplified animated LTS GIF that I prepared for the LTS README rewrite which attempts to communicate the very basics of the LTS plan in one image.

### CODE REVIEW

File: locale/en/blog/community/v5-to-v7.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+Each major version of Node.js has an active life of 6 months in what we are now calling "Current". During this period we ship most of the active work that goes in to the Node.js codebase except for some items that we reserve for the next major release. Node.js version 5 was first released in October last year, so its "Current" period ended in April this year. At the end of this 6 month period, something different happens for odd and even versioned release lines. The even versions turn in to LTS (Long-term Support) and receive another 30 months of life; this happened for version 4 in October last year and will happen for version 6 in October this year. The odd versions, however, don't get this extended life. Instead, as a transitionary measure, we provide another 2 months of support where we'll ensure that important fixes make it into that release line.
```
Comment (addaleax) [inline locale/en/blog/community/v5-to-v7.md]: @rvagg have you seen https://github.com/nodejs/LTS/pull/128?

**Comment by stevemao** [comment]
Nice! really like the gif!

File: locale/en/blog/community/v5-to-v7.md (hunk 1 above)
Comment (lpinca) [reply locale/en/blog/community/v5-to-v7.md]: @rvagg @addaleax I guess this should be updated to specify that additional support period is 3 months?

Comment (addaleax) [reply locale/en/blog/community/v5-to-v7.md]: This should be updated to specify whatever the actual support period is. It seems to me that @rvagg, @jasnell, and me all have different impressions on where we landed…

File: locale/en/blog/community/v5-to-v7.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+## We have a plan!
+
+![LTS Schedule Summary](/static/images/blog/201609_lts_schedule_summary.gif)
+
+Last year, the core team devised an LTS and release plan to balance the various wants and needs expressed by Node.js users. Chief among those were:
+
+1. Stability
+2. Progress
+
+The io.js diversion was useful for many reasons, including the opportunity we had to lean into this "progress" thing. We learned that there is a necessary trade-off between "stability" and the rapid iteration of the platform. Some of it was manageable but much was unavoidable. Breaking the entire C++ add-on ecosystem each time we upgraded V8 turned out to be quite painful for the Node.js package ecosystem. This is due to the heavy reliance on compiled native components in Node.js userland and the V8 project not having to be too concerned with the breakage and pain caused by the constant [API](wikipedia API) and [ABI](wikipedia ABI) churn since its primary consumer is Chromium.
```
Comment (ofrobots) [inline locale/en/blog/community/v5-to-v7.md]: Is it necessary to continue to sling mud at V8 here, especially given the recent history and the amount of energy myself and folks from V8 have been putting into better supporting Node?

Comment (ghost) [reply locale/en/blog/community/v5-to-v7.md]: i agree, this part sounds a bit _too_ harsh and personally biased :/

File: locale/en/blog/community/v5-to-v7.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+author: Rod Vagg
+---
+
+You may have missed it but at the end of June, the Node.js project said a final farewell to version 5. There will be no more patches, critical or otherwise, for this branch. To those who have been using Node.js for some time this may seem anomalous, shouldn't major versions stick around for _years_?
+
+## We have a plan!
+
+![LTS Schedule Summary](/static/images/blog/201609_lts_schedule_summary.gif)
+
+Last year, the core team devised an LTS and release plan to balance the various wants and needs expressed by Node.js users. Chief among those were:
```
Comment (zeke) [inline locale/en/blog/community/v5-to-v7.md]: `<abbr title="Long Term Support">LTS</abbr>` for this first instance of the acronym would be nice.

File: locale/en/blog/community/v5-to-v7.md (hunk 1 above)
Comment (rvagg) [reply locale/en/blog/community/v5-to-v7.md]: No, I hadn't, thanks for the link but see note there I just added.

File: locale/en/blog/community/v5-to-v7.md (hunk 2 above)
Comment (rvagg) [reply locale/en/blog/community/v5-to-v7.md]: Sorry, I didn't intend this as mud slinging. It's an objective observation based on a lot of history and accurately reflects the state we were in during io.js, even if things have improved lately. I'll adjust and ping you both when I have an update.

File: locale/en/blog/community/v5-to-v7.md (hunk 3 above)
Comment (rvagg) [reply locale/en/blog/community/v5-to-v7.md]: Cool, but perhaps I should just "Long-term Support (LTS)" here? I see I do "LTS (Long-term Support)" further down so I should remove that and put the expansion up here I suppose.

Comment (ghost) [reply locale/en/blog/community/v5-to-v7.md]: @rvagg yeah, that makes more sense

**Comment by rvagg** [comment]
Updated the post with feedback, put the "LTS" acronym expansion up front and adjusted the V8 API/ABI language to not suggest blame. @ofrobots @fene can you please check that language in the latest commit 1041f52d3f6 and sign-off? Happy to adjust more if you feel there's still a lingering negative suggestion there.

**Comment by rvagg** [comment]
For simplicity, here's the new text:

> Breaking the entire C++ add-on ecosystem each time we upgraded V8 turned out to be quite painful for the Node.js package ecosystem. This is due to the heavy reliance on compiled native components in Node.js userland and the difficulty Node.js has had in maintaining [API](wikipedia API) and [ABI](wikipedia ABI) stability while upgrading V8.

**Comment by ghost** [comment]
@rvagg i feel that now it's neutral enough 👍

**Comment by ofrobots** [comment]
Looks good!

**Comment by ghost** [comment]
general LGTM to this. is there a scheduled time to merge?

**Comment by misterdjules** [comment]
It was published to https://nodesource.com/blog/farewell-to-node-js-v5-preparing-for-v7/, which seems a bit confusing for a PR made to nodejs/nodejs.org.

**Comment by indutny** [comment]
This is a pretty strange way of merging `nodejs.org` PRs, @rvagg

**Comment by rvagg** [comment]
Intended as a cross-post on both blogs, they have different audiences and nodejs.org is likely to have broader reach anyway. Cross-post note will be added to the nodesource.com version now with this merged.

**Comment by rvagg** [comment]
https://nodejs.org/en/blog/community/v5-to-v7/

**Comment by jasnell** [comment]
One minor correction: you have the v4 LTS expiration as October 2018, when it should be April 2018. v6 LTS will expire April 2019.

**Comment by rvagg** [comment]
thanks for picking that up @jasnell, fixed!

**Comment by mhdawson** [comment]
LGTM.

**Pull request closed without merging**